api/users/views.py

```diff
@@ -574,3 +577,47 @@ def create(self, request, *args, **kwargs):
         user.requested_deactivation = True
         user.save()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class UserChangePassword(JSONAPIBaseView, generics.CreateAPIView, UserMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        CurrentUser,
+    )
+
+    required_read_scopes = [CoreScopes.NULL]
+    required_write_scopes = [CoreScopes.USER_SETTINGS_WRITE]
+
+    view_category = 'users'
+    view_name = 'user-password'
+
+    serializer_class = UserChangePasswordSerializer
+    throttle_classes = (SendEmailThrottle, )
+
+    def create(self, request, *args, **kwargs):
```

Comment:
@Johnetordoff Can you address this comment please?

Reply:
@sloria It's addressed [here](https://github.com/CenterForOpenScience/osf.io/pull/8610/files#diff-dd88c3f184d385cd8bef57c939c6d955R626) I can't call the typical logout() because it's working outside of flask's request context.